sirepo/package_data/static/js/raydata.js

```diff
@@ -316,6 +319,7 @@ SIREPO.app.directive('scansTable', function() {
                     </tr>
                   </tbody>
                 </table>
+                <div ng-if="awaitingScans">Loading scans...</div>
```

Comment:
I don't love this popping up and going away. What about a spinner that is just running as long as the we have the interval running?

Reply:
Do you mean regardless of whether it's actively awaiting a request for scans? I think there will be no way to tell whether there are no scans in the result or if it's just loading indefinitely.

Do you want to use the animation from loadingSpinner directive? Its style has its position set as absolute, so I will need to modify it or make a new directive if that's the one you would like to use.
Though to be honest I think the text looks cleaner and makes it clear what it's loading.

Reply:
> Do you mean regardless of whether it's actively awaiting a request for scans?

Yes. In my eyes (please push back if you disagree) the user doesn't care that a request is happening _right now_. What we're trying to let the user now is that when there are new scans to display we will show them. The GUI is "working behind the scenes" and they don't need to refresh the page to see something new. We happen to be searching on an interval every few seconds. But, when we switch to websockets maybe we won't poll. Updates will be pushed to the GUI whenever there are new scans. So, showing "loading scans..." isn't so important. What is important is that the user knows when there are new scans they will be displayed.  That process of "searching for new scans", as far as the user cares, is happening constantly. 

This may be worth opening a discuss issue because I could be in the minority but I also find things that show/hide on an interval to be distracting. For example, whenever someone has a GIF in a presentation slide I find it distracting that the GIF is just on a loop over and over while they talk.

Reply:
There are 2 pieces of information here: whether or not we are currently awaiting a request, and the fact that new scans matching the query will be displayed automatically.

Without the first piece, there is no way for a user to know for certain if there are no scans matching their query or if they're still awaiting a response. I think that's a frustrating experience for a user, and that's the benefit of the current "Loading...". An alternative solution would be to display something when the scan monitor returns an empty list, like "No scans to display". I still like the the loading indicator better.

The second piece is also useful information. I don't think a spinner is the best way to convey it though. 

I thought of a potential way to combine them though. Instead of popping up and going away, what if we had something that alternated between "Loading" and "Last updated at \<timestamp\>"? Here's a mockup I made with figma - text could change, just wanted to show the general idea. I'm not married to this but I think it's a possible solution.
![gifmock](https://user-images.githubusercontent.com/65698621/198156329-be9a63d6-903f-49c3-9352-032924c0389e.gif)

Reply:
You bring up some very good points. Thanks for creating that mockup. @moellep would you mind weighing in on the design?

To get you up to speed here is the problem we are working on: We have a table that we fill with data about analyses. Each row is one analysis and the columns are metadata about it (status, start time, stop time, etc). The query to get the data to populate the table takes long enough that we need some indication that it is happening. Likewise, we are polling on an interval for new data so the user doesn't have to refresh. On response there may be no data returned or there may be data. Do you have any thoughts on how to indicate all of this information to the user?

Reply:
@moellep not sure if you saw this

Reply:
Thanks - I'll take a look

Reply:
I agree that a spinner is distracting and plain text is fine. It is a bit confusing that it says "Loading scans..." or "Fetching scans..." but then does nothing when there are no scans, almost like the user is expecting something to get loaded.

I'm also not a fan of the text appearing and disappearing because it causes the panel border to constantly be changing height. This could be fixed by putting a place-holder black space in when the "loading scans.." is not shown. Something to keep the border from flipping back and forth.

Just a thought - how about show "Checking for new scans..." where the "..." changes every second, . . ...
And when it finds a new one, it could change to "Fetching scan" for a second, or until the new scan is available, and then immediately change back to "Checking for new scans..."?

Reply:
@rorour is there an issue to track this work?

Reply:
@e-carlin yes, https://github.com/radiasoft/raydata/issues/38

Reply:
Thanks!